dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +107,174 @@ public Bundle getBundleById(String id) throws DotDataException {
 	@Override
 	public void deleteBundle(String id) throws DotDataException {
 		bundleFactory.deleteBundle(id);
+	}
+
+	private void deleteBundleAndDependencies (final Bundle bundle, final User user) throws DotDataException {
+
+		final String bundleId = bundle.getId();
+		this.validateBundleDeletePermission(user, bundle);
+
+		Logger.info(this, "Removing bundle: " + bundleId + " and dependencies by: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Removing audit status for a bundle: " + bundleId);
+			this.publishAuditAPI.deletePublishAuditStatus(bundleId);
+		} catch (DotPublisherException e) {
+
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsFactory.deletePushedAssetsByBundle(bundleId);
+
+		Logger.info(this, "Removing all assets from bundle: " + bundleId);
+		this.bundleFactory.deleteAllAssetsFromBundle(bundleId);
+
+		Logger.info(this, "Removing environments from bundle: " + bundleId);
+		this.bundleFactory.deleteBundleEnvironmentByBundle(bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId);
+		this.deleteBundle(bundleId);
+	}
+
+	@WrapInTransaction
+	@Override
+	public void deleteBundleAndDependencies(final String bundleId, final User user) throws DotDataException {
+
+		final Bundle bundle = this.getBundleById(bundleId);
+		this.deleteBundleAndDependencies(bundle, user);
+	}
+
+	private void validateBundleDeletePermission(final User user, final Bundle bundle) throws DotDataException {
+
+		/**
+		if (!APILocator.getPermissionAPI().doesUserHavePermission(bundle,
+				PermissionAPI.PERMISSION_EDIT, user, false)) {
+
+			throw new DotSecurityException("User : " + user.getUserId() +
+					", is not allowed to delete the bundle: " + bundleId)
+		}*/
+		// todo: bundle is not a permissionable yet, so can not validate yet.
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		int offset               = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'offset' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)